### PR TITLE
Add create package and upload document methods

### PR DESCRIPTION
**Pull request by hawschiat**
### Changes
- Created `RequestBuilder`s classes and `Api` factory class to construct requests to OneSpan
- Added types for OneSpan's package and document endpoints
- Created `OneSpanSign` class and added methods to create package and upload documents

### Next Step
- Add more detailed typings
- Add unit tests, set up CI workflow on Github
- Add documentations

### CODE REVIEW

File: src/OneSpanSign.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      .withBody(payload)
+      .fetch();
+
+    return (await response.json()) as CreatePackageResponsePayload;
+  }
+
+  public async uploadDocument(
+    packageId: string,
+    payload: UploadDocumentRequestPayload,
+    documentBody: Buffer | ReadableStream
```
Comment (hawschiat) [inline src/OneSpanSign.ts:26]: Using `ReadableStream`, we could pipe the download stream from cloud storage servers (e.g. [Google Cloud Storage](https://cloud.google.com/storage/docs/streaming)) directly to the form data object

**Comment by hawschiat** [comment]
Putting this PR to draft temporarily to revise the typings based on the schema provided by OneSpan: https://community.onespan.com/sites/default/files/interactive_api_references/oss_1145.yaml

File: src/api/error.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Response } from 'node-fetch';
+
+export class HttpResponseError extends Error {
```
Comment (hawschiat) [inline src/api/error.ts:4]: Looking at OneSpan's error schema, we could improve this error class later on to set `code`, `messageKey` and `messages` as the class' public property

Screen Shot 2021-12-17 at 10 54 52 AM

File: src/OneSpanSign.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  CreatePackageRequestPayload,
+  CreatePackageResponsePayload,
+  DocumentMetadata,
+  UploadDocumentRequestPayload,
+} from './types';
+
+export class OneSpanSign {
+  constructor(private apiKey: string, private apiUrl: string) {}
+
+  public async createPackage(payload: CreatePackageRequestPayload): Promise<CreatePackageResponsePayload> {
```
Comment (francisco-jara) [inline src/OneSpanSign.ts:14]: wouldn't it make sense for this method to simply return a `Package`?

File: src/OneSpanSign.ts (hunk 1 above)
Comment (francisco-jara) [reply src/OneSpanSign.ts:26]: Nice 👍

File: src/OneSpanSign.ts (hunk 3 above)
Comment (hawschiat) [reply src/OneSpanSign.ts:14]: Unfortunately, this endpoint only returns an ID ([source](https://community.onespan.com/products/onespan-sign/sandbox#/Packages/api.packages.post))

Comment (francisco-jara) [reply src/OneSpanSign.ts:14]: ah poop

File: src/api/requestBuilders/BaseRequestBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    if (response.ok) return response;
+    throw new HttpResponseError(response);
+  }
+
+  public withAuthorizationHeader(value: string) {
+    this.requestHeaders.Authorization = value;
+    return this;
+  }
+
+  public accepts(value: string) {
```
Comment (francisco-jara) [inline src/api/requestBuilders/BaseRequestBuilder.ts]: should this be `withAcceptHeader` to be consistent with `withAuthorizationHeader`?

File: src/types/user.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import { External } from './external';
+import { Translation } from './translation';
+import { OptionalExclude } from './utils';
+
+// TODO: Type this
```
Comment (francisco-jara) [inline src/types/user.ts]: Do you wanna track these TODOs as Jiras or maybe GitHub issues?

File: src/types/utils.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export type OptionalExclude<T, Exclusion extends keyof T> = Required<Pick<T, Exclusion>> & Partial<Omit<T, Exclusion>>;
```
Comment (francisco-jara) [inline src/types/utils.ts:1]: Screen Shot 2021-12-17 at 12 27 42 PM

File: src/api/error.ts (hunk 2 above)
Comment (francisco-jara) [reply src/api/error.ts:4]: You'd put that in a OneSpan error specific class, right?

Comment (hawschiat) [reply src/api/error.ts:4]: Yeah, or just rename this class to something like `OneSpanError`, unless we also make calls to anything other than OneSpan's endpoints

File: src/types/user.ts (hunk 5 above)
Comment (hawschiat) [reply src/types/user.ts]: I think using GitHub issues (or projects) would be better to give visibility to all contributors!

File: src/types/utils.ts (hunk 6 above)
Comment (hawschiat) [reply src/types/utils.ts:1]: This utility type makes every except for the excluded property in the given type optional

Comment (francisco-jara) [reply src/types/utils.ts:1]: 👍 just j/k

**Review by francisco-jara — APPROVED** [review]
Good stuff!

**Pull request merged**